tests/ballerina-integration-test-utils/src/main/java/org/ballerinalang/test/context/ServerInstance.java

```diff
@@ -688,4 +689,11 @@ private String getPidWithLsof(int httpServerPort) throws BallerinaTestException
         }
         return pid;
     }
+
+    /**
+     * Reset server log reader.
+     */
+    public void resetServerLogReader() {
+        this.serverInfoLogReader = null;
```

Comment:
Shall we merge this with removeAllLeechers method? That way you have to call only one method to clean up log leechers.